### PR TITLE
Maks/IMD-6 first and last names validation

### PR DESCRIPTION
**Pull request by KovalenkoMaks**
## Description
Add validation for first and last names, which have to be not less then 2 symbols (Latin and Cyrillic letters) and/or a dash

## JIRA ticket
- [IMD-6](https://immortalv.atlassian.net/browse/IMD-6)

## Related PRs (Optional)
- https://github.com/immortalv/immortal_server/pull/23

[IMD-6]: https://immortalv.atlassian.net/browse/IMD-6?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/profiles/add-profile/steps/main-info.component.js [hunk 1]
```diff
@@ -29,9 +29,13 @@ function AddProfileMainInfo({ profile = {}, onSubmit, t }) {
 
     const { firstName, lastName, description } = state;
 
-    if (!firstName) return showErrorToast(t('ADD_PROFILE.ADD_FIRST_NAME'));
+    const nameValidation = (name) => {
+      return /^[a-zA-Zа-яА-ЯіІїЇєЄ']{2,}(-[a-zA-Zа-яА-ЯіІїЇєЄ']+)*$/.test(name);
```
Comment (andriiwp3) [inline src/components/profiles/add-profile/steps/main-info.component.js:33]: If you want to show that we expect only letters, we can use `/w`

Comment (KovalenkoMaks) [reply src/components/profiles/add-profile/steps/main-info.component.js]: fixed it

File: src/components/profiles/add-profile/steps/main-info.component.js [hunk 2]
```diff
@@ -29,9 +29,13 @@ function AddProfileMainInfo({ profile = {}, onSubmit, t }) {
 
     const { firstName, lastName, description } = state;
 
-    if (!firstName) return showErrorToast(t('ADD_PROFILE.ADD_FIRST_NAME'));
+    const nameValidation = (name) => {
+      return /[\wа-яА-ЯёЁіІїЇєЄ]{2,}(-[\wа-яА-ЯёЁіІїЇєЄ]+)*$/.test(name);
```
Comment (andriiwp3) [inline src/components/profiles/add-profile/steps/main-info.component.js:33]: if we use `\w' => we don't need the `a-яА-Я...` part and please don't mark conversations as resolved

**Review by andriiwp3 — CHANGES_REQUESTED** [review]

File: src/components/profiles/add-profile/steps/main-info.component.js (hunk 2 above)
Comment (andriiwp3) [reply src/components/profiles/add-profile/steps/main-info.component.js:33]: it was discussed during the meeting and we agreed to leave it as it is

**Pull request merged**